### DISCUSSION TITLE
Is there a simple way to print the Python function lambdify generates to file?

### DISCUSSION THREAD

**Discussion by moorepants** [sympy/sympy#25082] [Q&A] [answered]
I'd like to generate a file `func.py` that contains the Python function that lambdify generates, basically lambdify to file instead of immediately eval() it.

The desired behavior is something like this:

```
expr = a + b
lambdify_to_file([a, b], expr, name='func')
from func import func
func(2, 3)
```

Is there a simple way to access the string lambdify creates before it is eval'd? I know that I could copy the lambdify function and replace the eval lines with something that writes to file. But can I do this easily with the current SymPy API without having to write any new functions?

The reason I'd like this is that we create very large expressions for lambdify and I'd like to cache the function to disk so I don't always have to regenerate it (which can take 5 second to minutes). autowrap can write the generated files to disk for use, for example. I'm looking for similar behavior here or some other simple way to cache things. I may be able to pickle the resulting Python function from lambdify, so that's an option too, but I'd like to have the code on disk to be able to manipulate it if desired.

**Comment by moorepants** [comment]
Here is an explanation of how to do this with dill: https://stackoverflow.com/questions/29079923/save-load-sympy-lambdifed-expressions

**Reply by moorepants** [reply]
And: https://stackoverflow.com/questions/31314517/how-to-serialize-sympy-lambdified-function

**Comment by oscarbenjamin** [comment, marked as answer]
The lambdify dosctring shows how to get the code with `inspect.getsource`. It is unfortunately missing the imports so something like `from numpy import *` is needed.

**Reply by moorepants** [reply]
Perfect that works for me. Thanks!
